sdk/core/azure-core/src/main/java/com/azure/core/http/policy/UserAgentPolicy.java

```diff
@@ -59,11 +63,46 @@ public UserAgentPolicy(String userAgent) {
      * If the passed configuration contains true for AZURE_TELEMETRY_DISABLED the platform information won't be included
      * in the user agent.
      *
+     * @param applicationId User specified application Id.
+     * @param sdkName Name of the client library.
+     * @param sdkVersion Version of the client library.
+     * @param version {@link ServiceVersion} of the service to be used when making requests.
+     * @param configuration Configuration store that will be checked for the AZURE_TELEMETRY_DISABLED.
+     */
+    public UserAgentPolicy(String applicationId, String sdkName, String sdkVersion, Configuration configuration,
+                           ServiceVersion version) {
+        boolean telemetryDisabled = configuration.get(Configuration.PROPERTY_AZURE_TELEMETRY_DISABLED, false);
+        if (telemetryDisabled) {
+            this.userAgent = String.format(DISABLED_TELEMETRY_USER_AGENT_FORMAT, sdkName, sdkVersion,
+                version.getVersion());
+        } else {
+            if (applicationId == null) {
+                this.userAgent = String.format(USER_AGENT_FORMAT, sdkName, sdkVersion, getPlatformInfo(),
+                    version.getVersion());
+            } else {
+                this.userAgent = String.format(APPLICATION_ID_USER_AGENT_FORMAT, applicationId, sdkName, sdkVersion,
+                    getPlatformInfo(), version.getVersion());
+            }
+        }
+    }
+
+    /**
+     * Creates a UserAgentPolicy with the {@code sdkName} and {@code sdkVersion} in the User-Agent header value.
+     *
+     * If the passed configuration contains true for AZURE_TELEMETRY_DISABLED the platform information won't be included
+     * in the user agent.
+     *
+     * @deprecated This method is deprecated as it does not set the custom application id sent by user on the
+     * "User-Agent" header when sending a request.
+     * <p> Use {@link UserAgentPolicy#UserAgentPolicy(String, String, String, com.azure.core.util.Configuration,
```

Comment:
Get rid of the <p> and move the explanation text directly after the last sentence finishes.

Reply:
Removed the whole text as we decided to not deprecate the older constructor.